Common/Core/RecoDecay.h

```diff
@@ -724,12 +734,14 @@ class RecoDecay
           return -1;
         }
         // Check that the number of direct daughters is not larger than the number of expected final daughters.
-        if (particleMother.daughtersIds().back() - particleMother.daughtersIds().front() + 1 > static_cast<int>(N)) {
-          // Printf("MC Rec: Rejected: too many direct daughters: %d (expected %ld final)", particleMother.daughtersIds().back() - particleMother.daughtersIds().front() + 1, N);
-          return -1;
+        if constexpr (!checkProcess) {
+          if (particleMother.daughtersIds().back() - particleMother.daughtersIds().front() + 1 > static_cast<int>(N)) {
+            // Printf("MC Rec: Rejected: too many direct daughters: %d (expected %ld final)", particleMother.daughtersIds().back() - particleMother.daughtersIds().front() + 1, N);
+            return -1;
+          }
```

Comment:
@vkucera Sorry for jumping in, but isn't this check (which here we are turning off if `checkProcess` is true) redundant since re-done in https://github.com/AliceO2Group/O2Physics/blob/964ccfbc129a645503ab7edac69a873565168b30/Common/Core/RecoDecay.h#L739 ? What do you think?

Reply:
No it is not redundant.
The first check discards particles which have too many daughters before even checking how many are returned by `getDaughters`.
The second check discards the particles for which `getDaughters` did not return the expected number of daughters.